src/store/message-store-level.ts

```diff
@@ -89,7 +100,69 @@ export class MessageStoreLevel implements MessageStore {
       if (message) { messages.push(message); }
     }
 
-    return messages;
+    const sortedRecords = this.sortRecords(messages, sort);
+    return this.paginateRecords(sortedRecords, pagination);
+  }
+
+  private async paginateRecords(
+    messages: GenericMessage[],
+    pagination: Pagination
+  ): Promise<GenericMessage[]> {
+    const { messageCid: messageId, limit = 0 } = pagination;
+    if (messageId === undefined && limit > 0) {
+      return messages.slice(0, limit);
+    } else if (messageId === undefined) {
+      return messages; // return all
+    }
+
+    for (let i = 0; i < messages.length; i++) {
+      const testId = await Message.getCid(messages[i]);
+      if (testId === messageId && i + 1 < messages.length) {
+        const start = i + 1;
+        const end = limit === 0 ? undefined : limit + start;
+        return messages.slice(start, end);
+      }
+    }
+    return [];
+  }
+
+  /**
+   * This is a temporary naive sort, it will eventually be done within the underlying data store.
+   *
+   * This sort accepts a generalized MessageSort type which.
+   * currently only 3 type of search properties are explicitly used.
+   * `dateCreated`, `datePublished`, `messageTimestamp`
+   *
+   * If sorting is based on date published, records that are not published are filtered out.
+   * @param messages - Messages to be sorted if dateSort is present
+   * @param sort - Sorting scheme
+   * @returns Sorted Messages
+   */
+  private sortRecords(
+    messages: GenericMessage[],
+    sort: MessageSort
```

Comment:
After seeing this implementation I think your original ideas were better. If the implementor resorts into hardcoding the sortable index as was done here, we might as well define explicit interface to make expectation clear...

Perhaps something like:

export type MessageSort = {
  dateCreated?: SortOrder;
  datePublished?: SortOrder;
  messageTimestamp?: SortOrder;
};